### PR TITLE
fix several signed/unsigned comps, avoid sprintf in overlay code

### PR DESCRIPTION
**Pull request by sethalves**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c97eb20d6968c5a00ed9b11798a4160e793bfd31 available at: https://gist.github.com/8e38d0d6fffe54d31f53<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3d23446dcbd423696f72f2cb1ac699eac4b67ee7 available at: https://gist.github.com/ddf943bd6bdb2fb9df20<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 450787fde2177b71735d73897c39b4f39fe35193 available at: https://gist.github.com/03731535bc069bcff0be<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 450787fde2177b71735d73897c39b4f39fe35193 available at: https://gist.github.com/c6e1cc4005839be508b1<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

File: interface/src/Audio.h [hunk 1]
```diff
@@ -212,7 +212,7 @@ public slots:
     bool _outputStarveDetectionEnabled;
     quint64 _outputStarveDetectionStartTimeMsec;
     int _outputStarveDetectionCount;
-    int _outputStarveDetectionPeriodMsec;
+    uint _outputStarveDetectionPeriodMsec;
     int _outputStarveDetectionThreshold; // Maximum number of starves per _outputStarveDetectionPeriod before increasing buffer size
```
Comment (AndrewMeadows) [inline interface/src/Audio.h]: uint is not known on Mac.  Use uint32_t (from #include <stdint.h>) or quint32 from Qt.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a645d784d8a66bb1e4298449916689a8f1ece228 available at: https://gist.github.com/828ce25d35e38d2c967c<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bcf0846310dac8c72b4f5691f8db6b1f71a3cc65 available at: https://gist.github.com/a81e19cf4ae3c2ce5b4a<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: bcf0846310dac8c72b4f5691f8db6b1f71a3cc65 available at: https://gist.github.com/c20510d7ff8df83cd3f9<br/><br/>Installers available for download at:<br>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.exe<br>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4220.dmg

**Pull request merged**